waku/protocol/v1/waku_protocol.nim

```diff
@@ -85,13 +85,11 @@ type
     received: HashSet[Hash]
     accounting*: Accounting
 
-  P2PRequestHandler* = proc(peer: Peer, envelope: Envelope) {.gcsafe.}
-
   WakuNetwork = ref object
     queue*: ref Queue
     filters*: Filters
     config*: WakuConfig
-    p2pRequestHandler*: P2PRequestHandler
+    mailserver*: MailServer
```

Comment:
I don't think I like the current structure much. Mailserver should not become part of the waku_protocol like this.
The registration of the request handler allows you to split these off. Unless there are other issues that I currently am not seeing?

Imo the waku_mail.nim (or waku_mail_server if you like an extra file) should use (read: import) waku_protocol and not the other way around.

Reply:
I've done a very small refactor to highlight this better: https://github.com/status-im/nim-waku/pull/34